rust-client/src/main.rs

```diff
@@ -73,8 +77,20 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
 
     let opt = Opt::parse();
 
-    let mut client =
-        grpc::punchr_service_client::PunchrServiceClient::connect(opt.url.clone()).await?;
+    // Custom tls client config that accepts all certificates.
+    let tls = ClientConfig::builder()
+        .with_safe_defaults()
+        .with_custom_certificate_verifier(Arc::new(DummyVerifyAll))
```

Comment:
Why use `DummyVerifyAll` here?

Reply:
The default one somehow does not accept the certificate of out tests server. Not sure why, especially since it works for the go-clients. I will try to debug this, but as an short-term fix this was the easiest work-around.

Reply:
Do we need to merge here before investigating further why this is failing? While still defending against passive adversaries, this makes the client prone to active ones.

I understand that we don't see security as a high priority here. Still I would prefer to not merge a hack like the above into master.

Reply:
Solved now by setting an explicit ca certificate against which the server is verified. Tried to investigate why it does not work other, stumbled across https://github.com/briansmith/webpki/issues/20 which may be reason, but not 100% sure. Manually setting the ca certificate was the only solution I was able to come up with.
As default I hardcoded the root ca certificate for punchr.dtrautwein.eu, but a different one can be set via an CLI argument. Not happy about having it hardcoded, but it is more convenient than expecting the user to enter it every time. In general, I changed it now so that analogous to the go-client the server url defaults to https://punchr.dtrautwein.eu instead of requiring it as CLI argument.